RoddyCore/src/de/dkfz/roddy/config/ToolEntry.java

```diff
@@ -44,6 +44,17 @@ public void apply(BaseFile p) {
         public ToolConstraint clone() {
             return new ToolConstraint(onFailMethod, checkMethod);
         }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Well, it's a rule, not more. I think in a hash, the hashCode is used for the bucket, but within the bucket the equals() method is used to determine equality. Therefore, having different hashCodes but being equal(s) can lead to problems. hashCode() is kind of part of equality in the hash context.

But as I said, if we know, that we won't use the hashCode, we are safe.